### PR TITLE
dashboard: Make app history items actionable

### PR DESCRIPTION
**Pull request by jvatic**

### CODE REVIEW

**Comment by jvatic** [comment]
ping @titanous for review

**Comment by titanous** [comment]
@nobert Please try this branch out and let us know what you think.

**Comment by titanous** [comment]
How hard would it be to show a diff of the release in the deploy modal? It would be useful to see what env vars change.

**Comment by titanous** [comment]
The timestamp looks "wrong" when I deploy an older release. I'd expect the timestamp it was deployed, not the timestamp of when the release was created.

**Comment by jvatic** [comment]
- Deploy modal now shows env diff against current release.
- Fixed release timestamps to be the `created_at` of the event.

File: dashboard/app/lib/javascripts/dashboard/paginator.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	this.props = props;
+	this.Store = props.Store;
+	this.storeID = props.storeID;
+	this.wgs = [new WaitGroup(), new WaitGroup()]; // prevPage, nextPage
+	this.wgs[1].addOne();
+	this.__changeListeners = [];
+	this.handleStoreChange = this.handleStoreChange.bind(this);
+	this.Store.addChangeListener(this.storeID, this.handleStoreChange);
+	this.state = this.getInitialState();
+	window.p = this;
```
Comment (jvatic) [inline dashboard/app/lib/javascripts/dashboard/paginator.js]: This line needs removing

**Comment by jvatic** [comment]
- [ ] Fix issue where app history pagination gets stuck when loading next page

File: test/test_taffy_deploy.go [hunk 2]
```diff
@@ -26,17 +26,17 @@ func (s *TaffyDeploySuite) deployWithTaffy(t *c.C, app *ct.App, github map[strin
...
 		Meta: map[string]string{
-			"type":       "github",
-			"user_login": github["user_login"],
-			"repo_name":  github["repo_name"],
-			"ref":        github["ref"],
-			"sha":        github["sha"],
-			"clone_url":  github["clone_url"],
-			"app":        app.ID,
+			"github":    "true",
+			"user":      github["user"],
```
Comment (titanous) [inline test/test_taffy_deploy.go]: Maybe call this one `github_user`?

Comment (jvatic) [inline test/test_taffy_deploy.go]: And `repo` should be `github_repo` as it is also specific to github.

File: dashboard/app/lib/stylesheets/dashboard/github.scss [hunk 3]
```diff
@@ -330,3 +330,33 @@
     -webkit-flex-flow: row;
   }
 }
+
+.app-history-event > .pretty-radio {
```
Comment (jvatic) [inline dashboard/app/lib/stylesheets/dashboard/github.scss]: These rules should go in `app-history.scss`

**Comment by jvatic** [comment]
@titanous comments addressed

File: test/test_taffy_deploy.go [hunk 4]
```diff
@@ -50,46 +50,64 @@ func (s *TaffyDeploySuite) deployWithTaffy(t *c.C, app *ct.App, github map[strin
...
 	client := s.controllerClient(t)
 
 	github := map[string]string{
-		"user_login": "flynn-examples",
-		"repo_name":  "go-flynn-example",
-		"ref":        "master",
-		"sha":        "a2ac6b059e1359d0e974636935fda8995de02b16",
-		"clone_url":  "https://github.com/flynn-examples/go-flynn-example.git",
+		"user":      "flynn-examples",
+		"repo":      "go-flynn-example",
```
Comment (titanous) [inline test/test_taffy_deploy.go]: Do these match the new field names?

**Comment by titanous** [comment]
LGTM

File: test/test_taffy_deploy.go (hunk 4 above)
Comment (jvatic) [inline test/test_taffy_deploy.go]: This map is only used within the test.

**Pull request merged**